src/agent/src/namespace.rs

```diff
@@ -301,65 +299,20 @@ mod tests {
         assert_eq!(ns_root.persistent_ns_dir, tmpdir.path().to_str().unwrap());
     }
 
-    #[test]
-    fn test_namespace_type_get() {
-        #[derive(Debug)]
-        struct TestData<'a> {
-            ns_type: NamespaceType,
-            str: &'a str,
-        }
-
-        let tests = &[
-            TestData {
-                ns_type: NamespaceType::Ipc,
-                str: "ipc",
-            },
-            TestData {
-                ns_type: NamespaceType::Uts,
-                str: "uts",
-            },
-            TestData {
-                ns_type: NamespaceType::Pid,
-                str: "pid",
-            },
-        ];
-
-        // Run the tests
-        for (i, d) in tests.iter().enumerate() {
-            // Create a string containing details of the test
-            let msg = format!("test[{}]: {:?}", i, d);
-            assert_eq!(d.str, d.ns_type.get(), "{}", msg)
-        }
+    #[rstest]
+    #[case(NamespaceType::Ipc, "ipc")]
+    #[case(NamespaceType::Uts, "uts")]
+    #[case(NamespaceType::Pid, "pid")]
+    fn test_namespace_type_get(#[case] ns_type: NamespaceType, #[case] ns_name: &str) {
+        assert_eq!(ns_name, ns_type.get())
     }
 
-    #[test]
-    fn test_namespace_type_get_flags() {
-        #[derive(Debug)]
-        struct TestData {
-            ns_type: NamespaceType,
-            ns_flag: CloneFlags,
-        }
-
-        let tests = &[
-            TestData {
-                ns_type: NamespaceType::Ipc,
-                ns_flag: CloneFlags::CLONE_NEWIPC,
-            },
-            TestData {
-                ns_type: NamespaceType::Uts,
-                ns_flag: CloneFlags::CLONE_NEWUTS,
-            },
-            TestData {
-                ns_type: NamespaceType::Pid,
-                ns_flag: CloneFlags::CLONE_NEWPID,
-            },
-        ];
-
+    #[rstest]
+    #[case(NamespaceType::Ipc, CloneFlags::CLONE_NEWIPC)]
+    #[case(NamespaceType::Uts, CloneFlags::CLONE_NEWUTS)]
+    #[case(NamespaceType::Pid, CloneFlags::CLONE_NEWPID)]
```

Comment:
Optional suggestions: Maybe we could add a description to the cases to help identify which one might go wrong:
```
    #[case::Ipc(NamespaceType::Ipc, CloneFlags::CLONE_NEWIPC)]
    #[case::Uts(NamespaceType::Uts, CloneFlags::CLONE_NEWUTS)]
    #[case::Pid(NamespaceType::Pid, CloneFlags::CLONE_NEWPID)]
```
?

Reply:
Done.